lua/trunk.lua

```diff
@@ -40,6 +40,37 @@ local function findConfig()
 	return configDir and configDir .. "/.trunk/trunk.yaml" or ".trunk/trunk.yaml"
 end
 
+local function trim(s)
+	return s:match("^%s*(.-)%s*$")
+end
+
+local function getCliVersion()
+	local cmd = executionTrunkPath()
+	table.insert(cmd, #cmd + 1, "version")
+	local output = vim.fn.systemlist(cmd)
+	local possibleVersion = trim(output[#output])
+	if pcall(function()
+		vim.version.parse(possibleVersion)
+	end) then
+		return possibleVersion
+	else
+		-- version is not parsable
+		return nil
+	end
```

Comment:
pcall (protected call) is lua's version of try/catch - it calls the function it gets passed, returns true if it works, and false if there's an error.